kythe/go/extractors/cmd/bazel/bazel_go_extractor/bazel_go_extractor.go

```diff
@@ -109,352 +106,145 @@ func main() {
 }
 
 type extractor struct {
-	rules        vnameutil.Rules
-	toolArgs     *toolArgs
-	goos, goarch string
+	rules       vnameutil.Rules
+	compileArgs *compileArgs
+
+	goos, goarch, goroot string
+	cgoEnabled           bool
 }
 
 func (e *extractor) checkAction(_ context.Context, info *bazel.ActionInfo) error {
-	toolArgs, err := extractToolArgs(info.Arguments)
-	if err != nil {
-		return fmt.Errorf("extracting tool arguments: %v", err)
-	}
-	e.toolArgs = toolArgs
-
+	e.compileArgs = parseCompileArgs(info.Arguments)
 	for name, value := range info.Environment {
 		switch name {
 		case "GOOS":
 			e.goos = value
 		case "GOARCH":
 			e.goarch = value
+		case "GOROOT":
+			e.goroot = value
+		case "CGO_ENABLED":
+			e.cgoEnabled = value == "1"
```

Comment:
Does it have to be "1" or can it be any non-empty value?

Reply:
I'm not sure. `"1"` is the value the rules use now—and I assume that `"0"` and `""` could probably be taken as false. I didn't actually look at the implementation, though.